### PR TITLE
fix(FR-3039): resolve open CodeQL code-scanning alerts

### LINKED ISSUE #7716: Resolve open CodeQL code-scanning alerts (log redaction, ReDoS-safe filter parser, prototype-pollution guard, dead code removal)
Address the actionable open alerts on the GitHub code-scanning page (https://github.com/lablup/backend.ai-webui/security/code-scanning).

## Fixes in this issue

- **clear-text-storage-of-sensitive-data** (#115, #116, error/high) — `packages/backend.ai-client/src/client.ts` persists every request body to `localStorage['backendaiwebui.logs']`. Login already masks the password, but other secret-bearing requests (keypair `secret_key`, user create/update `password`, …) were stored in clear text. Add a recursive redaction pass that masks sensitive keys before the debug log is written.
- **polynomial-redos** (#71, #72, warning/high) — `packages/backend.ai-ui/src/components/BAIPropertyFilter.tsx` `parseFilterValue` used a lookahead regex with nested quantifiers vulnerable to catastrophic backtracking. Replace with a linear single-pass tokenizer that respects double-quoted spans.
- **prototype-polluting-assignment / prototype-pollution-utility** (#99, #100, #101, warning/medium) — `packages/backend.ai-ui/src/components/BAIGraphQLPropertyFilter.tsx` `buildNestedFilter` writes to keys derived from a dot path. Add a guard rejecting `__proto__` / `constructor` / `prototype` segments.
- **useless-regexp-character-escape / incomplete-html-attribute-sanitization** (#42, #46) — `src/lib/ansiup.js` is dead, unreferenced legacy code (not imported anywhere, not in the browser bundle). Delete the file, which resolves both alerts.

## Dismissed separately (no code change needed)

- docs-toolkit build-time alerts (#111, #118, #120, #121, #122) — operate on the project's own trusted markdown; not attacker-facing.
- #117 (`client-config.ts` trailing-slash regex) — bounded short input, negligible.
- #55, #56 (`src/lib/backend.ai-client-node.ts`) — wsproxy/Node-only module, not in the browser bundle.
- #36, #37, #39, #45 — already dismissed (deleted files / refactored-away SSRF).

## Verification

`bash scripts/verify.sh` → ALL PASS. BAIPropertyFilter (30) and BAIGraphQLPropertyFilter (3) vitest suites pass, including new regression + ReDoS-safety tests.

JIRA Issue: FR-3039

### PR DESCRIPTION
**Pull request by yomybaby**
Resolves #7716 (FR-3039)

Cleans up the actionable open alerts on the [code-scanning page](https://github.com/lablup/backend.ai-webui/security/code-scanning).

### `clear-text-storage-of-sensitive-data` — #115, #116 (error/high)
`packages/backend.ai-client/src/client.ts` persisted **every** request body to `localStorage['backendaiwebui.logs']`. Login already masked the password, but other secret-bearing requests (keypair `secret_key`, user create/update `password`, …) were written in clear text. Added a recursive `redactRequestParameters` pass that masks sensitive keys (`password`, `secret_key`, `token`, `otp`, …, matched case-insensitively) before the debug log is stored. Handles both JSON-string bodies (the common signed-request case) and already-parsed objects.

### `polynomial-redos` — #71, #72 (warning/high)
`parseFilterValue` in `BAIPropertyFilter.tsx` used `/\s+(?=(?:(?:[^"]*"){2})*[^"]*$)/` — a lookahead with nested quantifiers prone to catastrophic backtracking. Replaced with a linear single-pass tokenizer (`splitOutsideDoubleQuotes`) that respects double-quoted spans. Behavior is preserved (existing tests pass); added regression tests for quoted-whitespace / list values and a ReDoS-safety timing test.

### `prototype-polluting-assignment` / `prototype-pollution-utility` — #99, #100, #101 (warning/medium)
`buildNestedFilter` in `BAIGraphQLPropertyFilter.tsx` wrote to keys derived from a dot path. Added a guard that rejects `__proto__` / `constructor` / `prototype` path segments before building the nested object.

### dead code — #42, #46
`src/lib/ansiup.js` is unreferenced legacy code (not imported anywhere, not in the browser bundle). Deleted, which resolves both `useless-regexp-character-escape` and `incomplete-html-attribute-sanitization`.

## Not addressed here (dismissed with rationale)
- docs-toolkit build-time alerts #111 / #118 / #120 / #121 / #122 — operate on the project's own trusted markdown; not attacker-facing.
- #117 (`client-config.ts` trailing-slash regex) — bounded short input (an API URL); negligible.
- #55 / #56 (`src/lib/backend.ai-client-node.ts`) — wsproxy / Node-only module, not in the browser bundle.
- #36 / #37 / #39 / #45 — already dismissed (deleted files / refactored-away SSRF).

> Note: the `clear-text-storage` alerts (#115 / #116) may not auto-close on the next CodeQL run, since the static analyzer cannot model the runtime redaction. If they persist they should be dismissed as mitigated, referencing `redactRequestParameters`.

## Verification
- `bash scripts/verify.sh` → `=== ALL PASS ===` (Relay / Lint / Format / TypeScript).
- `BAIPropertyFilter` (30 tests) and `BAIGraphQLPropertyFilter` (3 tests) vitest suites pass, including the new regression + ReDoS-safety tests.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#7717** <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/7717?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/7717?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>flow:merge-queue</kbd> - adds this PR to the back of the merge queue
- <kbd>flow:hotfix</kbd> - for urgent changes, fast-track this PR to the front of the merge queue

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/lablup?ref=merge-queue-instructions-in-stack-comment&prId=7717).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend.ai-ui/src/components/BAIPropertyFilter.test.tsx [hunk 1]
```diff
@@ -56,6 +56,34 @@ describe('parseFilterValue', () => {
...
+
+  it('keeps quoted spans intact for list values', () => {
+    expect(
+      parseFilterValue('permission in ["READ_ONLY", "READ_WRITE"]'),
+    ).toEqual({
+      property: 'permission',
+      operator: 'in',
+      value: '["READ_ONLY", "READ_WRITE"]',
+    });
+  });
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply packages/backend.ai-ui/src/components/BAIPropertyFilter.test.tsx]: Good catch — replaced that duplicate with a genuinely new tokenizer edge case: a list whose elements contain whitespace *inside* the quotes (`name in ["READ ONLY", "READ WRITE"]`), which verifies the quoted-span handling rather than re-asserting the existing `in` test. Also folded in two further hardening tweaks from a parallel review: the tokenizer now matches the full `\s` class per-character (so Unicode whitespace like a non-breaking space still splits, still constant-time / no ReDoS), and `buildNestedFilter()` now also rejects empty path segments.
